tests/test_config.py

```diff
@@ -87,9 +132,10 @@ def test_config_merging():
     If we have both CLI args and a ``pyproject.toml`` file, the CLI args should
     have precedence.
     """
-    toml = StringIO(
-        dedent(
-            """\
+    toml = BytesIO(
```

Comment:
Could you factor this into a function, please?

```
def get_toml_bytes(config_string):
    return BytesIO(bytes(dedent(config_string), "utf-8"))
```

Reply:
I'm thinking of going with
```py
def get_toml_bytes(toml_str: str) -> BytesIO:
    """
    Wrap a string in BytesIO to play the role of the incoming config stream.
    """
    return BytesIO(bytes(toml_str, "utf-8"))
```
i.e. keeping `dedent` in the tests themselves. My thinking is that it then becomes clear why one is writing `"""\` rather than `"""`.

Reply:
Good catch!